{"uid":"73dd5c237549c458","name":"Verify that Search API will throw error supplier is not authorized to make Search","fullName":"features.APITests.steps.test_search_steps#test_verify_that_search_api_will_throw_error_supplier_is_not_authorized_to_make_search","historyId":"d63ae22c2fb90b9310fa587dcdb157a4","time":{"start":1759923837816,"stop":1759923839644,"duration":1828},"description":"/home/runner/work/imms_fhir_api_automation/imms_fhir_api_automation/features/APITests/search.feature: Verify that Search API will throw error supplier is not authorized to make Search","descriptionHtml":"<p>/home/runner/work/imms_fhir_api_automation/imms_fhir_api_automation/features/APITests/search.feature: Verify that Search API will throw error supplier is not authorized to make Search</p>\n","status":"failed","statusMessage":"AssertionError: \n Expected status code: 403, but got: 429. Response: {'message': 'This API is currently receiving a high volume of requests and is being rate limited.', 'scope': 'proxy', 'policy': 'quota', 'message_id': 'rrt-6937280167955954486-c-geu2-3709791-42515748-1'}","statusTrace":"fixturefunc = <function The_request_will_have_status_code at 0x7f5ef007b9c0>\nrequest = <FixtureRequest for <Function test_verify_that_search_api_will_throw_error_supplier_is_not_authorized_to_make_search[9449304424-COVID19]>>\nkwargs = {'context': <utilities.context.ScenarioContext object at 0x7f5eef48b150>, 'statusCode': '403'}\n\n    def call_fixture_func(\n        fixturefunc: _FixtureFunc[FixtureValue], request: FixtureRequest, kwargs\n    ) -> FixtureValue:\n        if is_generator(fixturefunc):\n            fixturefunc = cast(\n                Callable[..., Generator[FixtureValue, None, None]], fixturefunc\n            )\n            generator = fixturefunc(**kwargs)\n            try:\n                fixture_result = next(generator)\n            except StopIteration:\n                raise ValueError(f\"{request.fixturename} did not yield a value\") from None\n            finalizer = functools.partial(_teardown_yield_fixture, fixturefunc, generator)\n            request.addfinalizer(finalizer)\n        else:\n            fixturefunc = cast(Callable[..., FixtureValue], fixturefunc)\n>           fixture_result = fixturefunc(**kwargs)\n\nvenv/lib/python3.11/site-packages/_pytest/fixtures.py:898: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\ncontext = <utilities.context.ScenarioContext object at 0x7f5eef48b150>\nstatusCode = '403'\n\n    @then(parsers.parse(\"The request will be unsuccessful with the status code '{statusCode}'\"))\n    @then(parsers.parse(\"The request will be successful with the status code '{statusCode}'\"))\n    def The_request_will_have_status_code(context, statusCode):\n        print(context.response.status_code)\n        print(int(statusCode))\n>       assert context.response.status_code == int(statusCode), f\"\\n Expected status code: {statusCode}, but got: {context.response.status_code}. Response: {context.response.json()} \\n\"\nE       AssertionError: \nE        Expected status code: 403, but got: 429. Response: {'message': 'This API is currently receiving a high volume of requests and is being rate limited.', 'scope': 'proxy', 'policy': 'quota', 'message_id': 'rrt-6937280167955954486-c-geu2-3709791-42515748-1'}\n\nfeatures/APITests/steps/common_steps.py:66: AssertionError","flaky":false,"newFailed":false,"newBroken":false,"newPassed":false,"retriesCount":0,"retriesStatusChange":false,"beforeStages":[{"name":"setup_environment","time":{"start":1759923531411,"stop":1759923531411,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"stepsCount":0,"attachmentsCount":0,"hasContent":false},{"name":"global_context","time":{"start":1759923531413,"stop":1759923532948,"duration":1535},"status":"passed","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"stepsCount":0,"attachmentsCount":0,"hasContent":false},{"name":"context","time":{"start":1759923837816,"stop":1759923839203,"duration":1387},"status":"passed","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"stepsCount":0,"attachmentsCount":0,"hasContent":false},{"name":"_pytest_bdd_example","time":{"start":1759923839669,"stop":1759923839669,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"stepsCount":0,"attachmentsCount":0,"hasContent":false}],"testStage":{"description":"/home/runner/work/imms_fhir_api_automation/imms_fhir_api_automation/features/APITests/search.feature: Verify that Search API will throw error supplier is not authorized to make Search","status":"failed","statusMessage":"AssertionError: \n Expected status code: 403, but got: 429. Response: {'message': 'This API is currently receiving a high volume of requests and is being rate limited.', 'scope': 'proxy', 'policy': 'quota', 'message_id': 'rrt-6937280167955954486-c-geu2-3709791-42515748-1'}","statusTrace":"fixturefunc = <function The_request_will_have_status_code at 0x7f5ef007b9c0>\nrequest = <FixtureRequest for <Function test_verify_that_search_api_will_throw_error_supplier_is_not_authorized_to_make_search[9449304424-COVID19]>>\nkwargs = {'context': <utilities.context.ScenarioContext object at 0x7f5eef48b150>, 'statusCode': '403'}\n\n    def call_fixture_func(\n        fixturefunc: _FixtureFunc[FixtureValue], request: FixtureRequest, kwargs\n    ) -> FixtureValue:\n        if is_generator(fixturefunc):\n            fixturefunc = cast(\n                Callable[..., Generator[FixtureValue, None, None]], fixturefunc\n            )\n            generator = fixturefunc(**kwargs)\n            try:\n                fixture_result = next(generator)\n            except StopIteration:\n                raise ValueError(f\"{request.fixturename} did not yield a value\") from None\n            finalizer = functools.partial(_teardown_yield_fixture, fixturefunc, generator)\n            request.addfinalizer(finalizer)\n        else:\n            fixturefunc = cast(Callable[..., FixtureValue], fixturefunc)\n>           fixture_result = fixturefunc(**kwargs)\n\nvenv/lib/python3.11/site-packages/_pytest/fixtures.py:898: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\ncontext = <utilities.context.ScenarioContext object at 0x7f5eef48b150>\nstatusCode = '403'\n\n    @then(parsers.parse(\"The request will be unsuccessful with the status code '{statusCode}'\"))\n    @then(parsers.parse(\"The request will be successful with the status code '{statusCode}'\"))\n    def The_request_will_have_status_code(context, statusCode):\n        print(context.response.status_code)\n        print(int(statusCode))\n>       assert context.response.status_code == int(statusCode), f\"\\n Expected status code: {statusCode}, but got: {context.response.status_code}. Response: {context.response.json()} \\n\"\nE       AssertionError: \nE        Expected status code: 403, but got: 429. Response: {'message': 'This API is currently receiving a high volume of requests and is being rate limited.', 'scope': 'proxy', 'policy': 'quota', 'message_id': 'rrt-6937280167955954486-c-geu2-3709791-42515748-1'}\n\nfeatures/APITests/steps/common_steps.py:66: AssertionError","steps":[],"attachments":[{"uid":"6b9fa0a7dd1042bf","name":"Step Passed: Send a search request with GET method with invalid NHS Number '9449304424' and valid Disease Type 'COVID19'","source":"6b9fa0a7dd1042bf.txt","type":"text/plain","size":126},{"uid":"1e744481c8417604","name":"Step Failed: The request will be unsuccessful with the status code '403'","source":"1e744481c8417604.txt","type":"text/plain","size":348},{"uid":"bc48c8f8199b1a67","name":"stdout","source":"bc48c8f8199b1a67.txt","type":"text/plain","size":191}],"parameters":[],"shouldDisplayMessage":true,"stepsCount":0,"attachmentsCount":3,"hasContent":true},"afterStages":[],"labels":[{"name":"epic","value":"Immunization Service"},{"name":"suite","value":"Search the immunization of a patient"},{"name":"feature","value":"Search the immunization of a patient"},{"name":"tag","value":"Search_Feature"},{"name":"tag","value":"supplier_name_SONAR"},{"name":"parentSuite","value":"features.APITests.steps"},{"name":"host","value":"runnervmwhb2z"},{"name":"thread","value":"2457-MainThread"},{"name":"framework","value":"pytest"},{"name":"language","value":"cpython3"},{"name":"package","value":"features.APITests.steps.test_search_steps"},{"name":"resultFormat","value":"allure2"}],"parameters":[{"name":"_pytest_bdd_example","value":"{'NHSNumber': '9449304424', 'DiseaseType': 'COVID19'}"}],"links":[],"hidden":false,"retry":false,"extra":{"severity":"normal","retries":[],"categories":[{"name":"Product defects","matchedStatuses":[],"flaky":false}],"tags":["Search_Feature","supplier_name_SONAR"]},"source":"73dd5c237549c458.json","parameterValues":["{'NHSNumber': '9449304424', 'DiseaseType': 'COVID19'}"]}